### PR TITLE
Improve thread signals and slots

### LINKED ISSUE #213: Delete finished threads and remove callbacks from arguments
### Feature you want to implement
Currently `iquip` creates a lot of threads and do not manage them at all.
Moreover, we are passing the callback slots as arguments of thread constructors.
I think this restricts the signal-slot connection of threads, e.g., number of signals and slots.
Therefore, I want to make the threads more stable in terms of signal-slots.

### How the feature is implemented
- Connect `deleteLater()` to the `finished` signal.
- Remove callbacks from arguments and connect them outside the thread.

### Additional context
We may consider using [`QThreadPool`](https://doc.qt.io/qt-5/qthreadpool.html) but it uses [`QRunnable`](https://doc.qt.io/qt-5/qrunnable.html) instead of [`QThread`](https://doc.qt.io/qt-5/qthread.html) which is too abstract and not a `QObject` (so it does not have signals).

### PR DESCRIPTION
**Pull request by kangz12345**
This closes #213.

Since this has many changes (not just additions), it must be tested and reviewed thoroughly.
I tested this on the lab PC.

### CODE REVIEW

File: iquip/apps/builder.py [hunk 1]
```diff
@@ -356,22 +356,19 @@ def __init__(
         schedOpts: Dict[str, Any],
         ip: str,
         port: int,
-        callback: Callable[[int], None],
         parent: Optional[QObject] = None
     ):  # pylint: disable=too-many-arguments
         """Extended.
         
         Args:
             experimentPath, experimentArgs, schedOpts, ip, port: See the attributes section.
```
Comment (BECATRUE) [inline iquip/apps/builder.py]: Now, it is okay to write only "See the attributes section.".

File: iquip/apps/explorer.py [hunk 2]
```diff
@@ -161,6 +159,7 @@ def lazyLoadFile(self, experimentFileItem: QTreeWidgetItem):
             self._addFile,
             self
         )
+        self.fileFinderThread.finished.connect(self.fileFinderThread.deleteLater)
```
Comment (BECATRUE) [inline iquip/apps/explorer.py:162]: Please update `_FileFinderThread` call, and connect `fetched` signal to `self._addFile()`.

**Review by BECATRUE — CHANGES_REQUESTED** [review]

**Comment by kangz12345** [comment]
I tested the branch on the lab PC.

**Review by BECATRUE — APPROVED** [review]
LGTM!

**Pull request merged**